### ISSUE TITLE
Sentiment analysis

### ISSUE THREAD

**Issue by Nohyunsun** [Nohyunsun/ADHD-Self-Diagnosis-Across-Platforms#8]

**Comment by Nohyunsun** [comment]
🧹 5. Preprocessing
---
Typical steps:
- Lowercasing
- Remove URLs & mentions
- Remove emojis (or convert to text form)
- Normalization (“ㅋㅋㅋ” → “ㅋ”)
- Optional tokenization for word-level TF-IDF

Example:
```python
def clean_text(text):
    text = re.sub(r"http\S+|www\S+", "", text)
    text = re.sub(r"[^가-힣0-9A-Za-z\s]", "", text)
    return text.strip()
```

**Comment by Nohyunsun** [comment]
🧮 6. Feature Extraction
---
**Word-level TF-IDF**
```python
word_vectorizer = TfidfVectorizer(
    max_features=30000,
    ngram_range=(1,2),
    token_pattern=r"(?u)\b\w+\b"
)
```
**Character-level TF-IDF**
```python
char_vectorizer = TfidfVectorizer(
    analyzer='char',
    ngram_range=(1,3),
    max_features=35000
)
```

**Comment by Nohyunsun** [comment]
🤝 7. Model Training
---
Example with Logistic Regression:
```python
from sklearn.linear_model import LogisticRegression

clf_word = LogisticRegression(max_iter=500)
clf_char = LogisticRegression(max_iter=500)

ensemble = VotingClassifier(
    estimators=[
        ('word', clf_word),
        ('char', clf_char)
    ],
    voting='soft'
)

ensemble.fit(X_train_combined, y_train)
```
